### PR TITLE
Align the default version with Facebook business SDK

### LINKED ISSUE #18845: Facebook Ads Provider uses a deprecated version of the API
### Apache Airflow Provider(s)

facebook

### Versions of Apache Airflow Providers

2.0.1

### Apache Airflow version

2.1.1

### Operating System

Ubuntu 20.04

### Deployment

Docker-Compose

### Deployment details

_No response_

### What happened

Task fails because the hook uses a deprecated Facebook API version. The hook is calling v6.0 which is longer supported.

### What you expected to happen

I expected this task to connected to the Facebook API and fetch the requested data. 

My log files for the failed task output the following message:

```
facebook_business.exceptions.FacebookRequestError: 

  Message: Call was not successful
  Method:  POST
  Path:    https://graph.facebook.com/v6.0/act_1210763848963620/insights
  Params:  {'level': 'ad', 'date_preset': 'yesterday', 'fields': '["campaign_name","campaign_id","ad_id","clicks","impressions"]'}

  Status:  400
  Response:
    {
      "error": {
        "message": "(#2635) You are calling a deprecated version of the Ads API. Please update to the latest version: v11.0.",
        "type": "OAuthException",
        "code": 2635,
        "fbtrace_id": "AGRidwR5VhjU3kAJVUSkvuz"
      }
    }
```

Line 69 of https://github.com/apache/airflow/blob/main/airflow/providers/facebook/ads/hooks/ads.py should be changed to a newer API version.

### How to reproduce

Run the sample DAG posted here: https://airflow.apache.org/docs/apache-airflow-providers-google/stable/_modules/airflow/providers/google/cloud/example_dags/example_facebook_ads_to_gcs.html 

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by RyanSiu1995**
Upgrade the graph api to latest version
---
Closes #18845
No test needs to be updated because the parameters are supplied from caller

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/facebook/ads/hooks/ads.py
Comment (mik-laj) [inline airflow/providers/facebook/ads/hooks/ads.py]: It looks like a breaking change. Can you add a note in the changelog?

Comment (eladkal) [reply airflow/providers/facebook/ads/hooks/ads.py]: I think it would be better not to set our own default. We can import the default value from the library itself
https://github.com/facebook/facebook-python-business-sdk/blob/fbc3b6a5c2d54bd4184163b290dd614b668e4ee0/facebook_business/apiconfig.py#L22

or maybe better just not have a default value for `api_version` at all.
The upstream lib made this parameter optional:
https://github.com/facebook/facebook-python-business-sdk/blob/fbc3b6a5c2d54bd4184163b290dd614b668e4ee0/facebook_business/api.py#L181
That way it will use the default value of the upstream lib if the user didn't provide a value.

Comment (RyanSiu1995) [reply airflow/providers/facebook/ads/hooks/ads.py]: Instead of dropping the default value, why not making it as Optional?
User can still define their own version.
But the library will fall back to facebook_businss default api version if they don't specifiy it.

Comment (eladkal) [reply airflow/providers/facebook/ads/hooks/ads.py]: sorry if wasn't clear - this is what I meant.
Just treat it like any other optional param with default of `None`.

File: CHANGELOG.txt
Comment (eladkal) [inline CHANGELOG.txt]: This is not the place for it.
The note should be in the provider change log.
see https://github.com/apache/airflow/pull/18493/files for example

Comment (mik-laj) [reply CHANGELOG.txt]: I'm glad you want to write a new note. Remember that this note is intended for users.
Make sure it contains the following information:

- [ ] Previous behaviors
- [ ] New behaviors
- [ ] If possible, a simple example of how to migrate. This may include a simple code example.
- [ ] If possible, the benefit for the user after migration e.g. "we want to make these changes to unify class names."
- [ ] If possible, the reason for the change, which adds more context to that interest, e.g. reference for Airflow Improvement Proposal or external documentation.

More tips can be found in the guide:
https://developers.google.com/style/inclusive-documentation

Comment (potiuk) [reply CHANGELOG.txt]: For now - for provider changes we do NOT expect anyone to change any changelogs - until we introduce a tool like towncrier, making any rules like that makes more harm than good - mainly because we have:
a) no agreed rules (quoting google here is nice but I did not recall we took that as a base)
b) we have no agreement between committers on different rules
c) the release manager of provider (currently mysefl) prepares the changelog.

I do not think we have any rules, more I think introducing any rules, requires introducing protection and verification that those rules are followed (otherwise some changes will have it but others will not) - so unless we document the rules we should follow and have some way for the committers to make sure the commenrs are added, I think it makes little sense to ask people to follow such rules which are not documented/agreed.

I think for now it's best to remove the changelog entries, and leave it up to release manger, but it would be great if someone introduce automation and verification (I will do it it November, but maybe someone wants to do it earlier).

Comment (mik-laj) [reply CHANGELOG.txt]: So where do we want to describe the breaking changes? I am afraid that the title alone is not enough in many cases. The task manager also often does not have enough knowledge to prepare a description of the change that is sufficient, because the `git diff` often does not show the full context. The preparation of the note also forces us to reflect on whether we really need to introduce such a breaking change.(see example: https://github.com/apache/airflow/pull/18879#discussion_r726616811)
Regarding the content of the notes about breaking time, this has been discussed in the past, and we have some hints in the main UPDATING.md file.
https://github.com/apache/airflow/blob/ab5b2bfca6b5be57a19bdbfe2206319bd2e71ac9/UPDATING.md#L64-L78
Here is a related discussion: https://github.com/apache/airflow/pull/18879#discussion_r726616811

Comment (potiuk) [reply CHANGELOG.txt]: > So where do we want to describe the breaking changes? I am afraid that the title alone is not enough in many cases.

Sure - we need to figure it out. It's not done yet. We need:

1) a way to do it
2) a way to flag if not done for all PR
3) a documentation for contributors to follow 
4) a way to automate it to bring it to changelog automatically

Until it's not done, it's just random advice, because no-one (including other committers) will follow it.

 I think about implementing towncrier for that and documenting it, but until it is done, I think either someone does it (all points) or we just follow the currnet process (where release manager does it), because we have no documentation for it (unless you tell me otherwise and documentation is there)?

Comment (potiuk) [reply CHANGELOG.txt]: If you have other proposal, automation, documentation - I think it would be great to contribute it - but just advising in random PRs with random, not agreed, not consistent with other PRs advice brings more harm than good because it introduces terrible inconsistences and chaos.

Comment (mik-laj) [reply CHANGELOG.txt]: The first note in `CHANGELOG.rst` has been added by this PR. https://github.com/apache/airflow/pull/13767/files (file https://github.com/PolideaInternal/airflow/blob/28c96eec3cbd9954cb92995c7cfe356f2df847bb/airflow/providers/google/CHANGELOG.rst)
so we are now changing the rules that we previously established in that ticket. I was just trying to stick to what was agreed at the very beginning when we introduced the new changelogs for provider packages.

Comment (potiuk) [reply CHANGELOG.txt]: You refer to single change done by you in Jan. Since then we had 7 releases of providers  which followed different proces and every one of them  would have taken much longer time for coordination, automation if it was not automated how it is today. 

The thing you added then was a single case that did not establish a proces, way how to do it for scaled number of providers and how to make sure it is followed. As such - it does not really  establishes a "process".
It's an accidental change that is dificutlt to follow in scale, there is no automation to enforce it, nor maintain, and none of the committers is even aware what should they tell others when then review a change if they were to follow it. If it just "done" but not agreed, documented and neither "automation" is in place to kee'p it nor people involved know that it should be followed. 

Until this is done, this does not exist as a rule. It's just a "single change'.

It was ok for 1 provider and something you added and it was never documented how to do it.

Since then there were 6 or 7 waves of providers which were followed using different process which means I can release 60 providers in maye 1-2 hours. You are now trying to make a change to it wihich potentially changes it to 6 hours because relase manager will have to make extra checks on every single change and reconcile the changelog manually and make sure that every single change has been either manually added already or automatically incorporated into changelog.

If you reallyy want to fight for it and change the current process - I am perfectly ok with that - there is a next wave of providers to release in October. If you would like to take a lead on that and release them, I am all for changing it now and won't object changing the current approach. 

Are you ok with it? Can we count on it?

Comment (mik-laj) [reply CHANGELOG.txt]: I did not move the migration guides from a separate file to the changelogs. 
Screenshot 2021-10-12 at 02 43 27
This change has been committed by @potiuk on 1 Feb.

For the sake of clarity, I don't want to change the process now. I just wanted to understand what has changed since we made ["a one change"](https://github.com/apache/airflow/pull/13767/files) that has structured the package documentation until now.  Now I understand that we just gained experience and discovered the weaknesses of the current process. I'm totally fine with that.

Comment (mik-laj) [reply CHANGELOG.txt]: I think we now have a continuation of this thread. 
https://github.com/apache/airflow/pull/13767#discussion_r561478559
I have nothing to add, nor do I want to change anything in the process now. I am also not going to ask for manual editing in the CHANGELOG.rst file.

Comment (RyanSiu1995) [reply CHANGELOG.txt]: Hi all, I am going to remove that from CHANGELOG.rst and let it alone, right?

Comment (potiuk) [reply CHANGELOG.txt]: pleease :). I think we agreed with @mik-laj finally ;). Sorry for being dragged into such discussion. We are both passionate about it and sometimes I have to strong of an opinion :D

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**
